src/shared/utilities.js

```diff
@@ -82,13 +82,35 @@ export const fetchPost = (url, body) => (
 );
 
 /**
- *
+ * Creates the paramTypes for a Twitter specific search term.
+ * @param array of strings representing search paramaterTypes
+ * @returns array of objects representing search terms with meta data
+ */
+export const createTwitterParamTypes = (paramTypesText) => [{
+  name: 'author',
+  selected: paramTypesText.indexOf('author') > -1,
+  icon: getParamTypeIcon('author')
+}, {

```

Comment:
This could very easily be refactored into:
[makeParamType('author'), makeParamType('hashtag'), ...]
where
makeParamType = (type) => ({name: type, selected: etc etc